stage #4/module #5. AWS/theory/dynamodb.md

```diff
@@ -0,0 +1,143 @@
+[Read more](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/HowItWorks.CoreComponents.html#HowItWorks.CoreComponents.TablesItemsAttributes)
+
+There are terms that are referred to as Read Capacity Unit (RCU) and Write Capacity Unit (WCU). These define throughput of a particular DynamoDB table and directly influence the cost<br/>
+Read capacity unit (RCU): Each API call to read data from your table is a read request. Read requests can be strongly consistent, eventually consistent, or transactional. For items up to 4 KB in size, one RCU can perform one strongly consistent read request per second. Items larger than 4 KB require additional RCUs. For items up to 4 KB in size, one RCU can perform two eventually consistent read requests per second. Transactional read requests require two RCUs to perform one read per second for items up to 4 KB. For example, a strongly consistent read of an 8 KB item would require two RCUs, an eventually consistent read of an 8 KB item would require one RCU, and a transactional read of an 8 KB item would require four RCUs<br/>
+Write capacity unit (WCU): Each API call to write data to your table is a write request. For items up to 1 KB in size, one WCU can perform one standard write request per second. Items larger than 1 KB require additional WCUs. Transactional write requests require two WCUs to perform one write per second for items up to 1 KB. For example, a standard write request of a 1 KB item would require one WCU, a standard write request of a 3 KB item would require three WCUs, and a transactional write request of a 3 KB item would require six WCUs.
+[Read more](https://aws.amazon.com/dynamodb/pricing/provisioned/)
+
+Pretty important to be aware of the most common way to reach a table programmatically via API. Scan and query operations are available for these purposes.<br/>
+In general, scan works with the whole item, considering all the attributes it has while query looks up a certain partition based on partition/hash key
+[Ream more](https://dynobase.dev/dynamodb-scan-vs-query/#:~:text=Difference%20Between%20Query%20and%20Scan%20in%20DynamoDB&text=While%20Scan%20is%20%22scanning%22%20through,see%20the%20difference%20in%20speed.)
+
+DynamoDB also provides with a service that cashes information anf acts as an in-memory database. Amazon DynamoDB Accelerator (DAX) is a fully managed, highly available, in-memory cache for DynamoDB that delivers up to a 10x performance improvement.
+[Read more](https://aws.amazon.com/dynamodb/dax/)
+
+## Quiz
```

Comment:
@a1exanddrovich Please, place quiz in one page

Reply:
fixed